src/evergreen/api.py

```diff
@@ -321,6 +323,107 @@ def test_stats_by_project(self,
+
+        :param project_id: Id of patch to query for.
+        :param after_date: Collect stats after this date.
+        :param before_date: Collect stats before this date.
+        :param group_num_days: Aggregate statistics to this size.
+        :param requesters: Filter by requestors (mainline, patch, trigger, or adhoc).
+        :param tasks: Only include specified tasks.
+        :param variants: Only include specified variants.
+        :param distros: Only include specified distros.
+        :param group_by: How to group results (test_task_variant, test_task, or test)
+        :param sort: How to sort results (earliest or latest).
+        :return: Patch queried for.
+        """
+        params = {
+            'after_date': after_date,
+            'before_date': before_date,
+        }
+        if group_num_days:
+            params['group_num_days'] = group_num_days
+        if requesters:
+            params['requesters'] = requesters
+        if tasks:
+            params['tasks'] = tasks
+        if variants:
+            params['variants'] = variants
+        if distros:
+            params['distros'] = distros
+        if group_by:
+            params['group_by'] = group_by
+        if sort:
+            params['sort'] = sort
+        url = self._create_url('/projects/{project_id}/task_stats'.format(project_id=project_id))
+        task_stats_list = self._paginate(url, params)
+        return [TaskStats(task_stat, self) for task_stat in task_stats_list]
+
+    def task_reliability_by_project(self,
+                                    project_id,
+                                    after_date=None,
+                                    before_date=None,
+                                    group_num_days=None,
+                                    requesters=None,
+                                    tasks=None,
+                                    variants=None,
+                                    distros=None,
+                                    group_by=None,
+                                    sort=None):
+        """
+        Get task reliability scores.
+
+        :param project_id: Id of patch to query for.
+        :param after_date: Collect stats after this date.
+        :param before_date: Collect stats before this date, defaults to nothing.
+        :param group_num_days: Aggregate statistics to this size.
+        :param requesters: Filter by requesters (mainline, patch, trigger, or adhoc).
+        :param tasks: Only include specified tasks.
+        :param variants: Only include specified variants.
+        :param distros: Only include specified distros.
+        :param group_by: How to group results (test_task_variant, test_task, or test)
+        :param sort: How to sort results (earliest or latest).
+        :return: Patch queried for.
+        """
+        if after_date is None:
+            date = datetime.utcnow() - timedelta(days=180)
+            after_date = date.strftime('%Y-%m-%d')
```

Comment:
In `src/evergreen/util.py`, there is [code](https://github.com/evergreen-ci/evergreen.py/blob/32c3a9c5ac42e32f6b7c6ee816fcad5f043def7b/src/evergreen/util.py#L21-L28) to handle dates and datetimes. Can we use that so we will only have to worrry about how evergreen formats times and dates?